File: backend_pocketbase/scripts/seed_demo_data.py
import random
import requests
import os
import time
from pocketbase import PocketBase
from pocketbase.client import FileUpload

# CONFIGURATION
PB_URL = "http://127.0.0.1:8090"
ADMIN_EMAIL = "admin@example.com"  # À MODIFIER si nécessaire
ADMIN_PASS = "admin123456"       # À MODIFIER si nécessaire

# IMAGES LIBRES (Wikimedia Commons)
IMAGE_SAMPLES = {
    "food": "https://upload.wikimedia.org/wikipedia/commons/thumb/6/6d/Good_Food_Display_-_NCI_Visuals_Online.jpg/800px-Good_Food_Display_-_NCI_Visuals_Online.jpg",
    "music": "https://upload.wikimedia.org/wikipedia/commons/thumb/3/30/Musical_instruments_on_display_at_the_Musical_Instrument_Museum_in_Phoenix_AZ.jpg/800px-Musical_instruments_on_display_at_the_Musical_Instrument_Museum_in_Phoenix_AZ.jpg",
    "sport": "https://upload.wikimedia.org/wikipedia/commons/thumb/c/cd/Football_in_Berlin.jpg/800px-Football_in_Berlin.jpg",
    "soiree": "https://upload.wikimedia.org/wikipedia/commons/thumb/a/a9/Nightclub_party.jpg/800px-Nightclub_party.jpg",
    "expo": "https://upload.wikimedia.org/wikipedia/commons/thumb/4/4b/Louvre_Museum_Wikimedia_Commons.jpg/800px-Louvre_Museum_Wikimedia_Commons.jpg",
    "networking": "https://upload.wikimedia.org/wikipedia/commons/thumb/d/d5/Business_Meeting_Networking.jpg/800px-Business_Meeting_Networking.jpg"
}

pb = PocketBase(PB_URL)

def download_image(url):
    temp_path = "temp_image.jpg"
    try:
        response = requests.get(url, stream=True, timeout=10)
        if response.status_code == 200:
            with open(temp_path, 'wb') as f:
                for chunk in response:
                    f.write(chunk)
            return temp_path
    except Exception as e:
        print(f"Erreur téléchargement image : {e}")
    return None

def seed():
    try:
        # 1. AUTHENTICATION SUPERUSER
        pb.admins.auth_with_password(ADMIN_EMAIL, ADMIN_PASS)
        print("✅ Authentifié en tant que Superuser")

        # 2. CREATE USERS
        user_ids = []
        names = ["Alice", "Bob", "Charlie", "David", "Eve", "Frank", "Grace", "Heidi", "Ivan", "Judy"]
        for name in names:
            email = f"{name.lower()}@demo.com"
            try:
                # Vérifie si l'utilisateur existe déjà
                existing = pb.collection("users").get_list(1, 1, {"filter": f'email="{email}"'})
                if existing.items:
                    user_ids.append(existing.items[0].id)
                    continue

                user = pb.collection("users").create({
                    "username": name.lower() + str(random.randint(10, 99)),
                    "email": email,
                    "password": "password123",
                    "passwordConfirm": "password123",
                    "name": name,
                    "bio": f"Passionné de {random.choice(list(IMAGE_SAMPLES.keys()))} - Membre démo Good App."
                })
                user_ids.append(user.id)
                print(f"👤 Utilisateur créé : {name}")
            except Exception as e:
                print(f"⚠️ Erreur utilisateur {name}: {e}")

        # 3. CREATE EVENTS
        categories = list(IMAGE_SAMPLES.keys())
        locations = [
            ("Paris - Tour Eiffel", 48.8584, 2.2945),
            ("Paris - Le Marais", 48.8575, 2.3614),
            ("Paris - Canal St-Martin", 48.8741, 2.3631),
            ("Paris - Montmartre", 48.8867, 2.3431),
            ("Paris - Bastille", 48.8531, 2.3691)
        ]

        event_ids = []
        for i in range(25):
            cat = random.choice(categories)
            loc = random.choice(locations)
            creator = random.choice(user_ids)
            title = f"{cat.capitalize()} {random.choice(['Génial', 'VIP', 'Communautaire', 'Afterwork', 'Chill'])} #{i+1}"
            
            try:
                # Éviter les doublons par titre
                if pb.collection("events").get_list(1, 1, {"filter": f'title="{title}"'}).items:
                    continue

                # Téléchargement photo
                img_path = download_image(IMAGE_SAMPLES[cat])
                
                data = {
                    "title": title,
                    "description": f"Venez nombreux pour cet événement {cat} ! Un moment de partage inoubliable au coeur de {loc[0]}.",
                    "start_date": f"2026-05-{random.randint(10, 28)} {random.randint(10, 20)}:00:00",
                    "is_public": random.choice([True, True, False]), # Plus de publics que de privés
                    "creator": creator,
                    "location_name": loc[0],
                    "lat": loc[1] + (random.uniform(-0.01, 0.01)),
                    "long": loc[2] + (random.uniform(-0.01, 0.01)),
                }

                # Upload photo si téléchargée
                files = []
                if img_path:
                    files.append(("photos", open(img_path, "rb")))

                event = pb.collection("events").create(data, files=files)
                event_ids.append(event.id)
                print(f"🎉 Événement créé : {title}")
                
                if img_path: os.remove(img_path) # Nettoyage
            except Exception as e:
                print(f"⚠️ Erreur événement {title}: {e}")

        # 4. MEMBRE, LIKES & MESSAGES (Laisser le temps aux hooks de créer les conversations)
        print("⏳ Attente des hooks backend (conversations)...")
        time.sleep(2)

        for event_id in event_ids:
            try:
                # Likes aléatoires
                for _ in range(random.randint(2, 6)):
                    u_id = random.choice(user_ids)
                    try:
                        pb.collection("event_likes").create({"event": event_id, "user": u_id})
                    except: pass # Ignore si déja liké

                # Membres aléatoires
                for _ in range(random.randint(3, 8)):
                    u_id = random.choice(user_ids)
                    try:
                        pb.collection("event_members").create({
                            "event": event_id,
                            "user": u_id,
                            "status": random.choice(["accepted", "accepted", "pending"]),
                            "role": "member"
                        })
                    except: pass

                # Messages dans la conversation auto-créée
                convs = pb.collection("conversations").get_list(1, 1, {"filter": f'event="{event_id}"'})
                if convs.items:
                    conv_id = convs.items[0].id
                    # Seuls les membres "accepted" peuvent envoyer (sécurité hook)
                    accepted_members = pb.collection("event_members").get_list(1, 20, {"filter": f'event="{event_id}" && status="accepted"'})
                    for mem in accepted_members.items:
                        pb.collection("messages").create({
                            "conversation": conv_id,
                            "author": mem.user,
                            "content": random.choice([
                                "Hâte d'y être !", "C'est à quelle heure déjà ?", 
                                "Est-ce qu'on peut ramener des amis ?", "Trop cool ce projet !",
                                "Est-ce qu'il y aura à manger ?", "On se retrouve devant ?"
                            ])
                        })
                print(f"✅ Activité générée pour l'événement {event_id}")

            except Exception as e:
                print(f"⚠️ Erreur activité event {event_id}: {e}")

        print("\n🚀 SEED TERMINÉ AVEC SUCCÈS !")

    except Exception as e:
        print(f"❌ ERREUR GLOBALE : {e}")

if __name__ == "__main__":
    seed()

File: backend_pocketbase/scripts/seed_demo_data.js
import PocketBase from 'pocketbase';

const PB_URL = "http://127.0.0.1:8090";
const ADMIN_EMAIL = "thomas.spinner0@gmail.com";
const ADMIN_PASS = "Inscriptio&1";
const GLOBAL_PASS = "azertyuiop";

const pb = new PocketBase(PB_URL);

const PARIS = [
    { lat: 48.8566, lng: 2.3522, name: "Île de la Cité" },
    { lat: 48.8606, lng: 2.3376, name: "Louvre" },
    { lat: 48.8530, lng: 2.3499, name: "Saint-Germain" },
    { lat: 48.8619, lng: 2.3323, name: "Tuileries" },
    { lat: 48.8867, lng: 2.3431, name: "Montmartre" },
    { lat: 48.8529, lng: 2.3692, name: "Bastille" },
    { lat: 48.8738, lng: 2.2950, name: "Arc de Triomphe" },
    { lat: 48.8584, lng: 2.2945, name: "Tour Eiffel" },
    { lat: 48.8462, lng: 2.3464, name: "Luxembourg" },
    { lat: 48.8649, lng: 2.3800, name: "République" },
    { lat: 48.8708, lng: 2.3051, name: "Parc Monceau" },
    { lat: 48.8443, lng: 2.3740, name: "Gare de Lyon" },
    { lat: 48.8530, lng: 2.3566, name: "Quartier Latin" },
    { lat: 48.8660, lng: 2.3619, name: "Marais" },
    { lat: 48.8768, lng: 2.3594, name: "Canal Saint-Martin" },
    { lat: 48.8810, lng: 2.3589, name: "Belleville" },
    { lat: 48.8490, lng: 2.3903, name: "Nation" },
    { lat: 48.8396, lng: 2.3825, name: "Bercy" },
    { lat: 48.8322, lng: 2.3561, name: "Place d'Italie" }
];

const TITLES = [
    "Afterwork Rooftop", "Yoga Sunrise", "Tech Meetup", "Brunch du Dimanche",
    "Pique-Nique Géant", "Dégustation Vins", "Running Matinal",
    "Atelier Photo", "Soirée Jazz", "Networking Devs",
    "Expo Street Art", "Concert Acoustique", "Dîner Secret",
    "Cours de Cuisine", "Tournoi Ping-Pong", "Spectacle Impro",
    "Ciné Plein Air", "Balade Vélo", "Apéro Jeux"
];

const MSGS = ["Salut à tous !", "Hâte d'y être !", "Quelqu'un connaît l'adresse ?",
    "Je ramène des boissons !", "C'est à quelle heure ?", "On se retrouve sur place !"];

async function seed() {
    try {
        await pb.admins.authWithPassword(ADMIN_EMAIL, ADMIN_PASS);
        console.log("OK auth");

        // CLEAN
        for (const col of ["messages", "conversations", "event_likes", "event_members", "events", "users"]) {
            const recs = await pb.collection(col).getFullList();
            for (const r of recs) await pb.collection(col).delete(r.id).catch(() => {});
        }
        console.log("OK clean");

        // USERS
        const testUser = await pb.collection("users").create({
            "email": "test@goodapp.com", "password": GLOBAL_PASS, "passwordConfirm": GLOBAL_PASS,
            "name": "Compte Test", "username": "tester", "bio": "Testeur officiel"
        });
        console.log("OK test user");

        const creators = [];
        for (let i = 1; i <= 10; i++) {
            const u = await pb.collection("users").create({
                "email": `creator${i}@demo.com`, "password": GLOBAL_PASS, "passwordConfirm": GLOBAL_PASS,
                "name": `Creator ${i}`, "username": `creator_${i}`
            });
            creators.push(u);
        }
        console.log("OK 10 creators");

        const participants = [];
        for (let i = 1; i <= 20; i++) {
            const u = await pb.collection("users").create({
                "email": `participant${i}@demo.com`, "password": GLOBAL_PASS, "passwordConfirm": GLOBAL_PASS,
                "name": `Participant ${i}`, "username": `participant_${i}`
            });
            participants.push(u);
        }
        console.log("OK 20 participants");

        // EVENTS (19 total)
        const events = [];
        for (let ci = 0; ci < creators.length; ci++) {
            const count = (ci % 3) + 1;
            for (let j = 0; j < count; j++) {
                const idx = events.length;
                if (idx >= PARIS.length) break;
                
                const isPublic = idx % 2 === 0;
                const loc = PARIS[idx];
                const d = new Date();
                d.setDate(d.getDate() + 5 + idx * 2);
                d.setHours(18 + (idx % 4), 0, 0, 0);

                const ev = await pb.collection("events").create({
                    "title": `${TITLES[idx]} @ ${loc.name}`,
                    "description": `Rejoignez-nous pour ${TITLES[idx].toLowerCase()} à ${loc.name}. Ambiance garantie !`,
                    "start_date": d.toISOString().replace('T', ' ').substring(0, 19) + 'Z',
                    "is_public": isPublic,
                    "creator": creators[ci].id,
                    "location_name": `${loc.name}, Paris`,
                    "lat": loc.lat,
                    "lng": loc.lng,
                    "likes_count": 0,
                    "members_count": 0
                });
                events.push({ id: ev.id, title: ev.title || TITLES[idx], _isPublic: isPublic });

                // EXPLICIT DB REPAIRS IN CASE HOOKS FAIL/SWALLOW:
                await pb.collection("conversations").create({ "event": ev.id }).catch(() => {});
                
                await pb.collection("event_members").create({
                    "event": ev.id,
                    "user": creators[ci].id,
                    "role": "owner",
                    "status": "accepted"
                }).catch(() => {});
            }
        }
        const publicEvents = events.filter(e => e._isPublic);
        console.log(`OK ${events.length} events (${publicEvents.length} pub, ${events.length - publicEvents.length} prv)`);

        // JOINS (20 participants × 2)
        const joinMap = {};
        for (let pi = 0; pi < participants.length; pi++) {
            const p = participants[pi];
            const e1 = publicEvents[pi % publicEvents.length];
            const e2 = publicEvents[(pi + 1) % publicEvents.length];
            const joined = [e1.id];

            await pb.collection("event_members").create(
                { "event": e1.id, "user": p.id, "status": "accepted", "role": "member" }
            ).catch(() => {});

            if (e2.id !== e1.id) {
                joined.push(e2.id);
                await pb.collection("event_members").create(
                    { "event": e2.id, "user": p.id, "status": "accepted", "role": "member" }
                ).catch(() => {});
            }
            joinMap[p.id] = joined;

            for (const eid of joined) {
                await pb.collection("event_likes").create({ "event": eid, "user": p.id }).catch(() => {});
            }
        }
        console.log("OK joins");

        // LIKES SANS JOIN
        const allIds = events.map(e => e.id);
        let extraLikes = 0;
        for (let pi = 0; pi < participants.length; pi++) {
            const p = participants[pi];
            const joined = joinMap[p.id] || [];
            const available = allIds.filter(id => !joined.includes(id));
            const count = (pi % 2 === 0) ? 2 : 3;
            for (let k = 0; k < count && k < available.length; k++) {
                await pb.collection("event_likes").create(
                    { "event": available[(pi + k) % available.length], "user": p.id }
                ).catch(() => {});
                extraLikes++;
            }
        }
        console.log(`OK ${extraLikes} extra likes`);

        // TEST USER
        const testJoin = publicEvents[0];
        const testLike = events.find(e => e.id !== testJoin.id);
        await pb.collection("event_members").create(
            { "event": testJoin.id, "user": testUser.id, "status": "accepted", "role": "member" }
        ).catch(() => {});
        await pb.collection("event_likes").create({ "event": testJoin.id, "user": testUser.id }).catch(() => {});
        await pb.collection("event_likes").create({ "event": testLike.id, "user": testUser.id }).catch(() => {});
        console.log("OK test user actions");

        // COUNTERS
        for (const ev of events) {
            const mc = await pb.collection("event_members").getList(1, 1, { filter: `event="${ev.id}" && status="accepted"` });
            const lc = await pb.collection("event_likes").getList(1, 1, { filter: `event="${ev.id}"` });
            await pb.collection("events").update(ev.id, { "members_count": mc.totalItems, "likes_count": lc.totalItems }).catch(() => {});
        }
        console.log("OK counters");

        // MESSAGES
        try {
             const convs = await pb.collection("conversations").getFullList();
             for (const conv of convs) {
                 const mems = await pb.collection("event_members").getFullList({ filter: `event="${conv.event}" && status="accepted"` });
                 if (mems.length >= 2) {
                     for (let i = 0; i < 3; i++) {
                         await pb.collection("messages").create({
                             "conversation": conv.id, "author": mems[i % mems.length].user, "content": MSGS[i]
                         }).catch(() => {});
                     }
                 }
             }
             console.log("OK messages");
        } catch(e) {
             console.log("Messages warn:", e.message);
        }

        console.log("\nDONE!");
        console.log(`Users: 31 | Events: ${events.length} | Pub: ${publicEvents.length} | Prv: ${events.length - publicEvents.length}`);
        console.log(`Extra likes: ${extraLikes} | Password: ${GLOBAL_PASS}`);

    } catch (e) {
        console.error("FATAL:", e.message);
        console.error("Response data:", e.response?.data);
        console.error("Full error:", e);
    }
}

seed();
